src/libraries/Microsoft.Extensions.Configuration.Binder/src/Microsoft.Extensions.Configuration.Binder.ref.cs

```diff
@@ -27,14 +27,14 @@ public static partial class ConfigurationBinder
         [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("In case the type is non-primitive, the trimmer cannot statically analyze the object's type so its members may be trimmed.")]
         public static object? GetValue(this Microsoft.Extensions.Configuration.IConfiguration configuration, [System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.All)] System.Type type, string key) { throw null; }
         [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("In case the type is non-primitive, the trimmer cannot statically analyze the object's type so its members may be trimmed.")]
-        public static object? GetValue(this Microsoft.Extensions.Configuration.IConfiguration configuration, [System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.All)] System.Type type, string key, object defaultValue) { throw null; }
+        public static object? GetValue(this Microsoft.Extensions.Configuration.IConfiguration configuration, [System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.All)] System.Type type, string key, object? defaultValue) { throw null; }
         [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("In case the type is non-primitive, the trimmer cannot statically analyze the object's type so its members may be trimmed.")]
-        public static T? GetValue<[System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.All)] T>(this Microsoft.Extensions.Configuration.IConfiguration configuration, string key) { throw null; }
+        public static T? GetValue<T>(this Microsoft.Extensions.Configuration.IConfiguration configuration, string key) { throw null; }
```

Comment:
This should be reverted.

Reply:
These .ref.cs files are auto-generated by GenAPI based on the by Roslyn generated reference assembly and don't serve as an input to anything. They exist just to have a record of public API changes. Is this maybe wrong in the source assembly or it GenAPI that makes this in-correct?

Reply:
When opening the generated reference assembly with ILSpy, this is what I see:

```
[RequiresUnreferencedCode("In case the type is non-primitive, the trimmer cannot statically analyze the object's type so its members may be trimmed.")]
[return: Nullable(2)]
public static T GetValue<[Nullable(2)] [DynamicallyAccessedMembers(/*Could not decode attribute arguments.*/)] T>(this IConfiguration configuration, string key)
{
	throw null;
}
```

Reply:
@ViktorHofer if these files are just to serve as a reference for the public API (and aren’t used to produce a ref assembly), would using something like the PublicApiAnalyzer fit the use case better?

Reply:
> Looks like GenAPI is the culprit.

Can you log a GenAPI bug for this? These attributes need to be preserved in the ref file - they are important to the API.

Reply:
> @ViktorHofer if these files are just to serve as a reference for the public API (and aren’t used to produce a ref assembly), would using something like the PublicApiAnalyzer fit the use case better?

@jkoritzinsky PublicApiAnalyzer is quite verbose in comparison to GenAPI, there are two files, a .shipping.txt and a .nonshipping.txt. Also you need two of these files for each target framework if you want to track the API surface of every tfm. In comparison, GenAPI emits the same C# file that we have been looking at for the last years and a C# allows to encode public api differences based on i.e. the target framework. @ericstj and I were discussion this more than a year ago and we spoke with the Roslyn team to see if PublicApiAnalyzer could instead emit C# code but we didn't get very far with that discussion.